src/parser.rs

```diff
@@ -2195,15 +2208,32 @@ impl<'a> Parser<'a> {
 
     /// Parse a comma-separated list of 1+ SelectItem
     pub fn parse_projection(&mut self) -> Result<Vec<SelectItem>, ParserError> {
+        // BigQuery allows trailing commas, but only in project lists
+        // e.g. `SELECT 1, 2, FROM t`
+        // https://cloud.google.com/bigquery/docs/reference/standard-sql/lexical#trailing_commas
+        //
+        // This pattern could be captured better with RAII type semantics, but it's quite a bit of
+        // code to add for just one case, so we'll just do it manually here.
+        let old_value = self.options.trailing_commas;
+        self.options.trailing_commas |= dialect_of!(self is BigQueryDialect);
```

Comment:
I agree a RAAI style thing would be better but this is fine too for now

Reply:
Cool